src/test.rs

```diff
@@ -0,0 +1,95 @@
+use crate::{
+    debug, error, info, init_redis_connection, init_tracing, log_to_redis, trace, warn,
+    LogLocation, CLIENT_ID, REDIS_POOL,
+};
+use anyhow::{anyhow, Result};
+use bb8_redis::redis::cmd;
+use std::{net::IpAddr, path::Path};
+
+#[tokio::test]
+async fn test_init_redis_connection() {
+    let ip_addr = IpAddr::from([127, 0, 0, 1]);
+    let port = 6379;
+    let client_id = "test_connection".to_string();
+    let _ = init_redis_connection(ip_addr, port, client_id)
+        .await
+        .map_err(|e| e.to_string());
+    let pool = REDIS_POOL.read().unwrap();
+
+    assert!(pool.is_some());
+    let client_id = CLIENT_ID.read().unwrap();
+    assert_eq!(*client_id, "test_connection".to_string());
+}
+
+#[tokio::test]
+async fn test_log_to_redis() -> Result<()> {
+    let ip_addr = IpAddr::from([127, 0, 0, 1]);
+    let port = 6379;
+    let client_id = "test_log_to_redis".to_string();
+    let _ = init_redis_connection(ip_addr, port, client_id)
+        .await
+        .map_err(|e| e.to_string());
+    let log = "Successfully Logged".to_string();
+    log_to_redis(&log).await.unwrap();
+    let pool = REDIS_POOL.read().unwrap();
+    let client_id = CLIENT_ID.read().unwrap();
+
+    match pool.clone() {
+        Some(pool) => {
+            let mut conn = pool.get().await?;
+            let test = cmd("LRANGE")
+                .arg(client_id.clone())
+                .arg(0)
+                .arg(-1)
+                .query_async::<_, Vec<String>>(&mut *conn)
+                .await?;
+            let _ = cmd("RPOP")
+                .arg(client_id.clone())
+                .query_async::<_, ()>(&mut *conn)
+                .await?;
+            assert_eq!(test, vec!["Successfully Logged".to_string()]);
+            Ok(())
+        }
+        None => Err(anyhow!("Redis pool is not initialized")),
+    }
```

Comment:
Before or after (or at both timing) the test, please run a clean-up, for the redis. "test_log_to_redis", for more robust testing. In this case, `DEL test_log_to_redis` command would do I assume.

Reply:
For the test_log_to_redis, I originally RPOP the contents after I receive the Vec<String>/List but I'll change that to DEL. I think I need to do that for test_macros though.

Reply:
Done, for `test_log_to_redis,` after I get the value from the client Id key I delete `test_log_to_redis`, and for `test_macros` I do something similar except I do it from the beginning since the user might want to look at the results before running a new test. However, I wonder what will happen when running DEL if there is no key so I might change it

Reply:
Good job, but maybe I should have been less ambiguous here, I actually think the sequence should be
1. clean up (DEL)
2. log_to_redis
3. verify with LRANGE

Maybe this kind of discussion will be helpful to sync our thoughts. I agree with the first answer. https://softwareengineering.stackexchange.com/questions/395768/should-you-clean-up-your-db-after-integration-tests

Reply:
> However, I wonder what will happen when running DEL if there is no key so I might change it

For this, I encourage you to try testing with your redis-cli. Let's do it!

Reply:
Done! Running delete on a non-existent key doesn't seem to affect it! It just returns a 0 to indicate no key has been deleted and a 1 if a key has been deleted. Other than that this is the redis tests and yields same results every time.

<img width="463" alt="Screenshot 2023-11-30 at 3 06 15 PM" src="https://github.com/aicers/log-broker/assets/102562226/2322a37e-3a63-43ee-b3e9-37e4f702eaae">